man/man1/runmond.1
.TH RUNMOND 1NEMO "9 March 2009"
.SH NAME
runmond \- frontend to the rmond program
.SH SYNOPSIS
\fBrunmond\fP [parameter=value]
.SH DESCRIPTION
\fBrunmond\fP is a frontend to N-MODY, a parallel code for
collisionless N-body simulations in modified Newtonian dynamics
(see also  Ciotti, Nipoti, Londrillo 2006)
The input file is a standard NEMO snapshot, except that 
all masses are taken equal to that of the first particle.
All output is accumulated in a run directory, as specified
with \fBoutdir=\fP. See the detailed description in the
N-MODY users guide for their meaning. Particle positions and
velocities are dumped one per snapshot, and can be converted
back to NEMO files with \fImodysnap(1NEMO)\fP.
.SH PARAMETERS
The following parameters are recognized in any order if the keyword
is also given:
.TP 20
\fBin=\fP
Input snapshot. No default. All masses are assumed equal the
first one found in the snapshot. Conversion to the format
required by \fIrmond\fP is done by \fIsnapmody(1NEMO)\fP.
No Default. 
.TP
\fBoutdir=\fP
Run directory in which all files produced by N-MODY are placed.
No default.    
.TP
\fBout=\fP
Output snapshot file name if to 
convert the \fImoutXX.bin\fP files back to a nemo file. Unless
an absolute pathname is given, this file is placed in the 
\fBoutdir\fP. Default: not used.
.TP
\fBnr=\fP
some help [128]     
.TP
\fBnth=\fP
some help [64]     
.TP
\fBnph=\fP
some help [128]     
.TP
\fBlmax=\fP
some help [32]     
.TP
\fBmodel=\fP
reads particles on mout00.bin file (where n=id_new) **do
.TP
\fBmond_ind=\fP
0=Newtonian, 1=MOND, 2=deepMOND [1]    
.TP
\fBspl_ord=\fP
some help [2]     
.TP
\fBrmap=\fP
some help [1]     
.TP
\fBa0=\fP
some help [1.0]     
.TP
\fBdt_iter=\fP
some help [0.3]     
.TP
\fBscale=\fP
some help [1.0]     
.TP
\fBiter_max=\fP
some help [30]     
.TP
\fBtol=\fP
some help [5]     
.TP
\fBnew=\fP
some help [0]     
.TP
\fBid_new=\fP
index to read from for initial conditions [0]
.TP
\fBnout=\fP
some help [10]     
.TP
\fBiene=\fP
some help [5]     
.TP
\fBmrate=\fP
some help [10]     
.TP
\fBtmax=\fP
some help [10.0]     
.TP
\fBdt_min=\fP
some help [1.e-5]     
.TP
\fBcfl=\fP
some help [0.3]     
.TP
\fBlp_ord=\fP
some help [2]     
.SH EXAMPLES
.SH BENCHMARK
The default 2,048,000 particle distribution takes about 32 minutes to integrate 1 timeunit.
.SH SEE ALSO
.nf
snapmody(1NEMO), modysnap(1NEMO), snapshot(5NEMO)
.fi
N_MODY_userguide.pdf
.PP
Ciotti, Nipoti, Londrillo (2006ApJ...640..741C)
.PP
http://urania.bo.astro.it/nipoti/astrophysics/codes/nmody.htm 
.SH FILES

.SH AUTHOR
Peter Teuben
.SH UPDATE HISTORY
.nf
.ta +1.0i +4.0i
26-Feb-09	V1.0 Created	PJT
.fi


src/nbody/evolve/n_mody/runmond.c
/*
 *  RUNMOND:   run the mond program
 */

#include <nemo.h>

string defv[] = {
  "in=???\n         Input snapshot",
  "outdir=???\n     Run directory",
  "out=\n           Output snapshot if to convert back to NEMO?",
  "nr=128\n         some help",
  "nth=64\n         some help",
  "nph=128\n        some help",
  "lmax=32\n        some help",
  "model=0\n        reads particles on mout00.bin file (where n=id_new) **do not change**",
  "mond_ind=1\n     0=Newtonian, 1=MOND, 2=deepMOND",
  "spl_ord=2\n      some help",
  "rmap=1\n         some help",
  "a0=1.0\n         some help",
  "dt_iter=0.3\n    some help",
  "scale=1.0\n      some help",
  "iter_max=30\n    some help",
  "tol=5\n          some help",
  "new=0\n          some help",
  "id_new=0\n       index to read from for initial conditions",
  "nout=10\n        ** number of snapshots to output",
  "iene=5\n         some help",
  "mrate=10\n       some help",
  "tmax=10.0\n      ** end of integration",
  "dt_min=1.e-5\n   some help",
  "cfl=0.3\n        some help",
  "lp_ord=2\n       some help",
  "VERSION=0.2\n    10-mar-2009 PJT",
  NULL,
};

string usage="Frontend to the rmond program";

string cvsid="$Id$";

void nemo_main()
{
  int nr, nth, nph, lmax ;
  int model,mond_ind;
  real spl_ord, rmap  ;
  real a0,  dt_iter,scale;
  int iter_max;
  real tol;
  int new, id_new, nout,iene, mrate;
  real tmax, dt_min, cfl;
  int lp_ord;
  stream parstr;
  string infile = getparam("in");
  string outfile = getparam("out");
  string outdir = getparam("outdir");
  string exefile = "rmond";
  string parfile = "input.data";
  char dname[256], cmd[256];

  nr = getiparam("nr");
  nth = getiparam("nth");
  nph = getiparam("nph");
  lmax = getiparam("lmax");
  model = getiparam("model");
  mond_ind = getiparam("mond_ind");
  spl_ord = getrparam("spl_ord");
  rmap = getrparam("rmap");
  a0 = getrparam("a0");
  dt_iter = getrparam("dt_iter");
  scale = getrparam("scale");
  iter_max = getiparam("iter_max");
  tol = getrparam("tol");
  new = getiparam("new");
  id_new = getiparam("id_new");
  nout = getiparam("nout");
  iene = getiparam("iene");
  mrate = getiparam("mrate");
  tmax = getrparam("tmax");
  dt_min = getrparam("dt_min");
  cfl = getrparam("cfl");
  lp_ord = getiparam("lp_ord");




  make_rundir(outdir);
  sprintf(dname,"%s/%s",outdir,parfile);

  parstr = stropen(dname,"w");

  fprintf(parstr,"nr, nth, nph, lmax \n");
  fprintf(parstr,"%d,  %d,  %d,  %d \n",nr, nth, nph, lmax);

  fprintf(parstr,"model,mond_ind,spl_ord, rmap\n");
  fprintf(parstr,"%d, %d, %g, %g\n",model,mond_ind,spl_ord, rmap);

  fprintf(parstr,"a0,  dt_iter,scale\n");
  fprintf(parstr,"%g, %g, %g\n",a0,  dt_iter,scale);

  fprintf(parstr,"iter_max, tol\n");
  fprintf(parstr,"%d, %g\n",iter_max, tol);

  fprintf(parstr,"new, id_new, nout,iene,mrate\n");
  fprintf(parstr,"%d, %d, %d, %d, %d\n",new, id_new, nout,iene,mrate);
  fprintf(parstr,"tmax, dt_min, cfl, lp_ord\n");
  fprintf(parstr,"%g, %g, %g, %d\n",tmax, dt_min, cfl, lp_ord);
  fprintf(parstr,"! --------------\n");
  fprintf(parstr,"! this file was produced automatically by NEMO::runmond\n");
  strclose(parstr);

  sprintf(cmd,"snapmody in=%s out=%s/%s",infile,outdir,"mout00.bin");
  run_program(cmd);

  goto_rundir(outdir);
  sprintf(cmd,"rmond");
  run_program(cmd);

  if (hasvalue("out")) {
    sprintf(cmd,"cat mout??.bin | modysnap - %s",outfile);
    run_program(cmd);
  }

}

goto_rundir(string name)
{
  dprintf(0,"CHDIR: %s\n",name);
  if (chdir(name))
    error("Cannot change directory to %s",name);
}

make_rundir(string name)
{
  dprintf(0,"MKDIR: %s\n",name);
  if (mkdir(name, 0755))
    warning("Run directory %s already exists",name);
}

run_program(string cmd)
{
  dprintf(0,"EXEC: %s\n",cmd);
  system(cmd);
}


